src/main/java/com/example/music/backend/user/service/UserServiceImpl.java

```diff
@@ -44,4 +51,11 @@ public void processOAuthPostLogin(OAuth2User oAuth2User) {
             userRepository.save(newUser);
         }
     }
+
+    @Override
+    public UserDto getDto(String email) {
```

Comment:
Keep union strategy. Service should return entity or DTO/Response classes. Don't mix, please